seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/source/ClickhouseSourceFactory.java

```diff
@@ -37,11 +61,60 @@ public String factoryIdentifier() {
         return "Clickhouse";
     }
 
+    @Override
+    public <T, SplitT extends SourceSplit, StateT extends Serializable>
+            TableSource<T, SplitT, StateT> createSource(TableSourceFactoryContext context) {
+        ReadonlyConfig readonlyConfig = context.getOptions();
+        List<ClickHouseNode> nodes = ClickhouseUtil.createNodes(readonlyConfig);
+
+        String sql = readonlyConfig.get(SQL);
+        ClickHouseNode currentServer = nodes.get(ThreadLocalRandom.current().nextInt(nodes.size()));
+        try (ClickHouseClient client = ClickHouseClient.newInstance(currentServer.getProtocol());
+                ClickHouseResponse response =
+                        client.connect(currentServer)
+                                .format(ClickHouseFormat.RowBinaryWithNamesAndTypes)
+                                .query(modifySQLToLimit1(sql))
+                                .executeAndWait()) {
+            TableSchema.Builder builder = TableSchema.builder();
+            int columnSize = response.getColumns().size();
+            for (int i = 0; i < columnSize; i++) {
+                String columnName = response.getColumns().get(i).getColumnName();
+                SeaTunnelDataType<?> seaTunnelDataType =
+                        TypeConvertUtil.convert(response.getColumns().get(i));
+                builder.column(
+                        PhysicalColumn.of(
+                                columnName, seaTunnelDataType, Long.MAX_VALUE, true, null, null));
```

Comment:
PhysicalColumn.of(
                                                column.getColumnName(),
                                                TypeConvertUtil.convert(column),
                                                Long.valueOf(column.getEstimatedLength()),
                                                column.getScale(),
                                                column.isNullable(),
                                                null,
                                                null);

Isn't that better?